sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/DataSourceStrategySuite.scala

```diff
@@ -311,6 +312,21 @@ class DataSourceStrategySuite extends PlanTest with SharedSparkSession {
     assert(PushableColumnAndNestedColumn.unapply(Abs('col.int)) === None)
   }
 
+  test("SPARK-36644: Push down boolean column filter") {
+    testTranslateFilter('col.boolean, Some(sources.EqualTo("col", true)))
+
+    val t = "test_table"
+    withTable(t) {
+      import testImplicits._
+      Seq(Some(true), Some(false), None).toDF().write.saveAsTable(t)
+      val df = spark.table(t)
+      df.where("value").queryExecution.executedPlan.collectFirst {
```

Comment:
We should also verify the result after pushdown.

Reply:
updated and passed all tests